### PR TITLE
[FLINK-6608] [security, config] Relax Kerberos login contexts parsing

### PR DESCRIPTION
**Pull request by tzulitai**
This PR allows whitespaces in the list of configured Kerberos login contexts.
It generally covers some over scenarios as well (covered in the additional parsing test in `SecurityUtilsTest`).

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityUtils.java [hunk 1]
```diff
@@ -230,10 +231,21 @@ private void validate() {
 		}
 
 		private static List<String> parseList(String value) {
-			if(value == null) {
+			if(value == null || value.isEmpty()) {
 				return Collections.emptyList();
 			}
-			return Arrays.asList(value.split(","));
```
Comment (sunjincheng121) [inline flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityUtils.java]: I think can we using regular expression: `return Arrays.asList(value.replaceAll("\\s*,\\s*", ",").split(","));`

Comment (tzulitai) [inline flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityUtils.java]: Ah, right, thanks for the review and suggestion :) Will use regex instead.

Comment (tzulitai) [inline flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityUtils.java]: Addressed!

Comment (EronWright) [inline flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityUtils.java]: An alternative to consider:
`Collections.list(new StringTokenizer(",X, ,,Y , Z,", ", "))` produces `[X, Y, Z]`.

Comment (sunjincheng121) [inline flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityUtils.java]: @EronWright your suggestion will be work. But I'd like using regular expression. The JDK DOC also has the same recommend:
 `StringTokenizer is a legacy class that is retained for compatibility reasons although its use is discouraged in new code. It is recommended that anyone seeking this functionality use the split method of String or the java.util.regex package instead.`
Please see: http://docs.oracle.com/javase/7/docs/api/java/util/StringTokenizer.html
What do you think ? @EronWright @tzulitai 
Best,
SunJincheng

Comment (EronWright) [inline flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityUtils.java]: The wisdom of that paragraph has been debated extensively on stackoverflow.  I'll just say that the `StringTokenizer` is not deprecated and, in my opinion, fits this scenario uniquely well.   Go ahead either way.

File: flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityUtils.java [hunk 2]
```diff
@@ -230,10 +230,15 @@ private void validate() {
...
+
+			return Arrays.asList(value
+				.replaceAll("\\s*,\\s*", ",") // remove whitespaces surrounding commas
+				.replaceAll(",,", ",") // remove empty entries
```
Comment (sunjincheng121) [inline flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityUtils.java]: Hi, @tzulitai `.replaceAll("\\s*,\\s*", ",").replaceAll(",,", ",") ` can not deal with `" a, b,,, c d, e "`
I think we can using the expression as follows:
 `str.trim().replaceAll("(\\s*,+\\s*)+", ",")` OR `str.replace("/^\\s+|\\s+$/g","").replaceAll("(\\s*,\\s*)+", ",");`

Comment (tzulitai) [inline flink-runtime/src/main/java/org/apache/flink/runtime/security/SecurityUtils.java]: Thanks again, that makes sense :)

**Comment by tzulitai** [comment]
Thanks for the reviews @sunjincheng121 @EronWright.
I personally prefer to stick with the regex approach. But really either way is fine :)

Don't really expect any test errors on this, but doing one final run just to be safe, and then merging.

**Comment by sunjincheng121** [comment]
+1 to merged.

**Comment by tzulitai** [comment]
Merging to 1.3 and master

**Pull request closed without merging**